docs/quickstart.rst

```diff
@@ -144,3 +144,111 @@ Otherwise, models are represented as dicts.
+* define a custom renderer that is able to deal with this serialization problem
+
+:mod:`pyramid_swagger` (starting from version X.X.X) provides:
+
+* a new renderer, called ``pyramid_swagger``
+* a new renderer factory of factories, called ``pyramid_swagger.renderer.PyramidSwaggerRendererFactory``
+
+Using this new solutions you can reduce your efforts on dealing with formatting and focus more on your business logic.
+
+-----------------------------------------
+How :mod:`pyramid_swagger` renderer works
+-----------------------------------------
+
+The new :mod:`pyramid_swagger` renderer is a wrapper around the default ``pyramid.renderers.JSON`` renderer.
+
+:mod:`pyramid_swagger` will receive, from your pyramid view, the object that has to be rendered, perform the marshaling operations and then call the default JSON renderer.
+
+.. note::
+    The usage of this renderer will allow you to better use the Custom Formats configurable with ``pyramid_swagger.user_formats``.
+
+
+Let' assume that your view returns ``{'date': datetime.date.today()}`` and that your response spec is similar to
+
+.. code-block:: json
+
+    {
+        "200": {
+            "description": "HTTP/200",
+            "schema": {
+                "properties": {
+                    "date": {
+                        "type": "string",
+                        "format": "date"
+                    }
+                }
+            }
+        }
+    }
+
+
+If your view is configured to use ``json`` renderer then your endpoint will surprisingly return HTTP/500 errors.
+The errors are caused by the fact that ``pyramid.renderers.JSON`` is not aware on how to convert a ``datetime.date`` object.
+
+If your view is configured to use ``pyramid_swagger`` renderer then your endpoint will provide HTTP/200 responses similar
+to ``{"date": "2017-09-16"}``.
+
+This is possible because the marsharling of the view return value converts the ``datetime.date`` object to its ISO 8601
+string representation that could be handled by the default JSON renderer.
+
+.. note::
+
+    The marshaling operation will be performed according to the specific response schema defined for the particular endpoint.
+    It means that if your response doesn't specify a field it will transparently moved to the wrapped renderer.
+
+----------------------------------------------
+How :mod:`PyramidSwaggerRendererFactory` works
+----------------------------------------------
+
+To increase the flexibility of the new rendering feature, it is exposed ``PyramidSwaggerRendererFactory`` class which will allow you to
+connect your custom defined renderer after the marshaling operation of the view result.
+
+The defined renderer will operate on the marshaled, according to the Swagger Specification, response.
+
+Example of definition of a custom renderer
+
+.. code-block:: python
+
+    class MyPersonalRendererFactory(object):
```

Comment:
I'm not sure this is a factory at all...just looks like a renderer to me?

Reply:
I copied / got inspired by https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/renderers.html#adding-a-new-renderer .